internal/app/machined/pkg/controllers/k8s/kubelet_spec.go

```diff
@@ -163,7 +166,7 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
 			return fmt.Errorf("error merging arguments: %w", err)
 		}
 
-		kubeletConfig, err := NewKubeletConfiguration(cfgSpec.ClusterDNS, cfgSpec.ClusterDomain, cfgSpec.ExtraConfig, cfgSpec.DefaultRuntimeSeccompEnabled)
+		kubeletConfig, err := NewKubeletConfiguration(cfgSpec)
```

Comment:
do i need to pass as?
```suggestion
		kubeletConfig, err := NewKubeletConfiguration(cfgSpec.DeepCopy())
```

Reply:
the map gets modified here: https://github.com/siderolabs/talos/pull/6021/files#diff-c1756580a1897c92944f52223cd04cfd6279aabfafc71c7bfcd30a0904e6b807R323

Reply:
looks like it's fine? as we don't use `cfgSpec` further, and what we get back as a resource is always a copy